### PR TITLE
[NFC] Refactor `TritonGENToLLVM`

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 1]
```diff
@@ -1415,8 +1372,50 @@ struct TritonMatrix2DBlockPrefetchLowering
   LogicalResult
   matchAndRewrite(TritonGEN::Matrix2DBlockPrefetchOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    LLVM::CallOp callOp = createGenISA2DBlockPrefetch(op, rewriter);
-    rewriter.replaceOp(op, callOp);
+    // TODO: Remove GenISA lowering after PoC productization is completed.
+    char *env = std::getenv("TRITONGEN_FORCE_GENISA");
+    const bool useGenISA = env ? (bool)std::atoi(env) : false;
```
Comment (victor-eds) [inline third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:1377]:
```suggestion
    const bool useGenISA = std::getenv("TRITONGEN_FORCE_GENISA");
```

Maybe we can just do this if the variable's defined?

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 2]
```diff
@@ -1330,7 +1232,63 @@ struct TritonMatrix2DBlockLoadLowering
       return success();
     }
 
-    rewriter.replaceOp(op, createGenISA2DBlockRead(op, rewriter));
+    // TODO: Remove GenISA lowering after PoC productization is completed.
+    char *env = std::getenv("TRITONGEN_FORCE_GENISA");
+    const bool useGenISA = env ? (bool)std::atoi(env) : false;
```
Comment (victor-eds) [inline third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:1237]: Same

**Review by victor-eds — APPROVED** [review]
Way cleaner now. Thanks, Whitney!

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp]: Good idea.

**Review by victor-eds — APPROVED** [review]

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:1377]: @victor-eds with the suggested change `TRITONGEN_FORCE_GENISA=0` would make `useGenISA` to be true, which is not what we want, so reverted the change.

**Pull request merged**